src/main/lrsql/admin/interceptors/xapi_credentials_override.clj

```diff
@@ -0,0 +1,24 @@
+(ns lrsql.admin.interceptors.xapi-credentials-override
+  (:require
+   [lrsql.input.auth :as auth-input]
+   [lrsql.ops.query.auth :as auth-q]
+   [lrsql.util :as util]
+   [next.jdbc :as jdbc]))
+
+(defn auth-by-cred-id-interceptor [lrs]
+  {:name ::auth-by-cred-id-interceptor
+   :enter (fn auth-by-cred-id-interceptor [ctx]
+            (let [modded-ctx
+                  (when-let [cred-id (get-in ctx [:request :params :credentialID])]
+                    (let [conn (-> lrs :connection :conn-pool)
+                          backend (:backend lrs)
+                          input (auth-input/query-credential-by-id-input cred-id)
+                          {api-key :api_key secret-key :secret_key}
+                          (jdbc/with-transaction [tx conn]
+                            (auth-q/query-credential-by-id backend tx input))]
+                      (when (and api-key secret-key)
+                        (let [base64 (util/str->base64encoded-str (str api-key ":" secret-key))]
+                          (-> ctx (update-in [:request :params] dissoc :credentialID)
+                              (assoc-in [:request :headers "authorization"]
+                                        (str "Basic " base64)))))))]
+              (or modded-ctx ctx)))})
```

Comment:
This approach is elegant, but: am I right in thinking that this would allow _any_ client to make such a request? I thought this was just for the admin UI front end?

Reply:
@invaliduser can you confirm / explain how normal JWT authentication is being applied in this approach? Rereading I am becoming concerned that there is a sec hole here, in line with what @milt mentioned.

Reply:
This would enable any client to do so, yeah.  They *would* have to have the (db-side) primary key of the keypair in question; the only place that is exposed is through `/admin/env`.  However, in principle, there's nothing to stop someone from making an authorized call to `env` and then plugging that into curl (or whatever)